models/orderDao.js

```diff
@@ -0,0 +1,95 @@
+const { appDataSource } = require("./dbconfig");
+
+const ORDER_STATUS = Object.freeze({
+  상품준비중: 1,
+  배송중: 2,
+  배송완료: 3,
+});
+
+const createOrder = async (
+  orderNumber,
+  totalPrice,
+  userId,
+  plantId,
+  plantQuantity,
+  potId,
+  potQuantity,
+  nutrientId,
+  nutrientQuantity
+) => {
+  const queryRunner = appDataSource.createQueryRunner();
+  await queryRunner.connect();
+  await queryRunner.startTransaction();
+
+  try {
+    await queryRunner.query(
+      `INSERT INTO orders (
+        order_number,
+        total_price,
+        user_id,
+        order_status_id
+      ) VALUES (?, ?, ?, ?);
+     `,
+      [orderNumber, totalPrice, userId, ORDER_STATUS.상품준비중]
+    );
+
+    const [orderId] = await queryRunner.query(
+      `SELECT
+      orders.id,
+      orders.total_price
+      FROM orders 
+      WHERE order_number = ?;
+      `,
+      [orderNumber]
+    );
+
+    await queryRunner.query(
+      `INSERT INTO order_products (
+        plant_id,
+        plant_quantity,
+        pot_id,
+        pot_quantity,
+        nutrient_id,
+        nutrient_quantity,
+        order_id
+      ) VALUES (?, ?, ?, ?, ?, ?, ?);
+      `,
+      [
+        plantId,
+        plantQuantity,
+        potId,
+        potQuantity,
+        nutrientId,
+        nutrientQuantity,
+        orderId.id,
+      ]
+    );
```

Comment:
typeORM의 query method를 통해 insert into를 하면 사실 결과값에는 객체구조로 여러 값들이 담깁니다. 그 중 `insertId`를 활용하시면 불필요한 쿼리를 없앨 수 있을 것 같습니다. 아래 코드 참고해주세요!

(또한 기존의 orders.id를 가져오는 query에서 total_price는 가져올 필요가 없었는데 들어가있었네요)
```suggestion
    const cresateOrder = await queryRunner.query(
      `INSERT INTO orders (
        order_number,
        total_price,
        user_id,
        order_status_id
      ) VALUES (?, ?, ?, ?);
     `,
      [orderNumber, totalPrice, userId, ORDER_STATUS.상품준비중]
    );

    await queryRunner.query(
      `INSERT INTO order_products (
        plant_id,
        plant_quantity,
        pot_id,
        pot_quantity,
        nutrient_id,
        nutrient_quantity,
        order_id
      ) VALUES (?, ?, ?, ?, ?, ?, ?);
      `,
      [
        plantId,
        plantQuantity,
        potId,
        potQuantity,
        nutrientId,
        nutrientQuantity,
        createOrder.insertId,
      ]
    );
```